tests/e2e/scale/test_cephfs_many_files.py

```diff
@@ -0,0 +1,238 @@
+    return return_list
+
+
+class MillionFilesOnCephfs(object):
+    """
+    Create pvc and cephfs pod, make sure that the pod is running.
+    """
+    def __init__(self):
+        self.cephfs_pvc = helpers.create_pvc(
+            constants.DEFAULT_STORAGECLASS_CEPHFS,
+            pvc_name=PVC_NAME,
+            namespace=DEFAULT_NS,
+            size=SIZE
+        )
+        self.cephfs_pod = helpers.create_pod(
+            interface_type=constants.CEPHFILESYSTEM,
+            pvc_name=self.cephfs_pvc.name,
+            namespace=DEFAULT_NS,
+            node_name='compute-0',
+            pod_name=POD_NAME
+        )
+        helpers.wait_for_resource_state(self.cephfs_pod, "Running", timeout=300)
+        logging.info("pvc and cephfs pod created")
+
+    def cleanup(self):
+        self.cephfs_pod.delete()
+        self.cephfs_pvc.delete()
+        logging.info("Teardown complete")
+
+
+@pytest.fixture(scope='session')
+def million_file_cephfs(request):
+    million_file_cephfs = MillionFilesOnCephfs()
+
+    def teardown():
+        million_file_cephfs.cleanup()
+    request.addfinalizer(teardown)
+
+
+@scale
+@ignore_leftovers
+@pytest.mark.parametrize(
+    argnames=["resource_to_delete", "build_temp"],
+    argvalues=[
+        pytest.param(
+            *['mgr', True],
+        ),
+        pytest.param(
+            *['mon', False],
+        ),
+        pytest.param(
+            *['osd', False],
+        ),
+        pytest.param(
+            *['mds', False],
+        ),
+    ]
+)
+class TestMillionCephfsFiles(E2ETest):
+    """
+    Million cephfs files tester.
+    """
+    def test_scale_million_cephfs_files(
+        self,
+        million_file_cephfs,
+        resource_to_delete,
+        build_temp
+    ):
+        """
+        Add a million files to the ceph filesystem
+        Delete each instance of the parametrized ceph pod
+        Once the ceph cluster is healthy, verify that no files were lost
+
+        args:
+            million_file_cephfs -- fixture
+            resource_to_delete (str) -- resource deleted for each testcase
+            build_temp -- Add one million files, if True.
+        """
+        if build_temp:
+            self.sample_list = add_million_files()
+        else:
+            with open(JSON_FNAME, 'r') as fd:
+                self.sample_list = json.load(fd)
+        proc = subprocess.Popen(
+            f'oc -n {DEFAULT_NS} rsh {POD_NAME} df | grep {TARGET_DIR}',
+            shell=True,
+            stdout=subprocess.PIPE
+        )
+        dfoutput = proc.communicate()[0]
+        logging.info(f"Df results on ceph pod - {dfoutput}")
+        if resource_to_delete in ['mgr', 'mon', 'osd', 'mds']:
+            logging.info(f"Testing respin of {resource_to_delete}")
+            disruption = disruption_helpers.Disruptions()
+            disruption.set_resource(resource=resource_to_delete)
+            no_of_resources = disruption.resource_count
+            for i in range(0, no_of_resources):
```

Comment:
Do we need this? At any point, we'd be deleting just one resource. isn't it?

Reply:
I think that you are correct.  I will remove the for loop and just have one call.